ui/src/utils/timeSeriesTransformers.js

```diff
@@ -198,6 +198,19 @@ export const filterTableColumns = (data, fieldNames) => {
   return filteredData[0].length ? filteredData : [[]]
 }
 
+export const orderTableColumns = (data, fieldNames) => {
+  const fieldsSortOrder = fieldNames.map(fieldName => {
+    return _.findIndex(data[0], dataLabel => {
+      return dataLabel === fieldName.internalName
+    })
+  })
+  const filteredFieldSortOrder = _.filter(fieldsSortOrder, f => f !== -1)
```

Comment:
consider combining the getting of the sort order and removing the columns that aren't in there by using fieldNames.reduce instead of map and then only getting and adding the index for fields that are visible so you only have to loop through once